addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/ElmahScanRuleUnitTest.java

```diff
@@ -0,0 +1,250 @@
+                    + "</body></html>";
+
+    private static final String VALID_RESPONSE_WITH_CONTENT =
+            "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n"
+                    + "<html><head>\n"
+                    + "<title>Error Log</title>\n"
+                    + "</head><body>\n"
+                    + "<h1>Error Log for testing</h1>\n"
+                    + "<p>Blah blah blah.</p>\n"
+                    + "</body></html>";
+
+    private static final String VALID_RESPONSE_WITHOUT_CONTENT =
+            "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n"
+                    + "<html><head>\n"
+                    + "<title>Something else</title>\n"
+                    + "</head><body>\n"
+                    + "<h1>Possibly a custom 404</h1>\n"
+                    + "<p>Blah blah blah.</p>\n"
+                    + "</body></html>";
+
+    @Override
+    protected ElmahScanRule createScanner() {
+        ElmahScanRule rule = new ElmahScanRule();
+        rule.setConfig(new ZapXmlConfiguration());
+        return rule;
+    }
+
+    @Test
+    public void shouldSendReasonableNumberOfMessagesInLowStrength() throws Exception {
+        // Given
+        rule.setAttackStrength(Plugin.AttackStrength.LOW);
+        rule.init(getHttpMessage(URL), parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent, hasSize(lessThanOrEqualTo(NUMBER_MSGS_ATTACK_PER_PAGE_LOW)));
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    public void shouldSendReasonableNumberOfMessagesInMediumStrength() throws Exception {
+        // Given
+        rule.setAttackStrength(Plugin.AttackStrength.MEDIUM);
+        rule.init(getHttpMessage(URL), parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent, hasSize(lessThanOrEqualTo(NUMBER_MSGS_ATTACK_PER_PAGE_MED)));
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    public void shouldSendReasonableNumberOfMessagesInHighStrength() throws Exception {
+        // Given
+        rule.setAttackStrength(Plugin.AttackStrength.HIGH);
+        rule.init(getHttpMessage(URL), parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent, hasSize(lessThanOrEqualTo(NUMBER_MSGS_ATTACK_PER_PAGE_HIGH)));
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    public void shouldSendReasonableNumberOfMessagesInInsaneStrength() throws Exception {
+        // Given
+        rule.setAttackStrength(Plugin.AttackStrength.INSANE);
+        rule.init(getHttpMessage(URL), parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(
+                httpMessagesSent, hasSize(lessThanOrEqualTo(NUMBER_MSGS_ATTACK_PER_PAGE_INSANE)));
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    public void shouldNotAlertIfNonExistingElmahFileReturnsNon404Code() throws Exception {
```

Comment:
Merge this with the following and call it `shouldNotAlertIfNon200CodeStdThreshold`? (and test with e.g. 401, 403, 404, 500?) Move closer to `shouldAlertIfForbiddenResponseLowThreshold`?